homeassistant/components/group/entity.py

```diff
@@ -437,15 +450,38 @@ def _async_update_group_state(self, tr_state: State | None = None) -> None:
             self._state = None
             return
         # If the entity domains have more than one
-        # on state, we use STATE_ON/STATE_OFF
+        # on state, we use STATE_ON/STATE_OFF, unless there is
+        # only one specific `on` state in use for one specific domain
+        elif (
+            len(active_on_states := _active_on_states()) == 1
```

Comment:
Maybe a generator that shorts out as soon as there is more than one?

Reply:
done